plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-catalog-controller.php

```diff
@@ -270,4 +276,89 @@ private function canonicalize_fields( array $fields ) {
 		sort( $fields, SORT_STRING );
 		return $fields;
 	}
+
+	/**
+	 * Check if AsyncGenerator from Product Feed plugin is available.
+	 *
+	 * @return bool True if AsyncGenerator is available, false otherwise.
+	 */
+	private function is_async_generator_available() {
+		return defined( 'WPFOAI_VERSION' ) && class_exists( '\Automattic\WooCommerce\ProductFeedForOpenAI\Integrations\POSCatalog\AsyncGenerator' );
+	}
+
+	/**
+	 * Request products catalog using AsyncGenerator (async processing).
+	 *
+	 * @param array $fields         Product/variation fields to include in the catalog.
+	 * @param bool  $force_generate Whether to force regenerate the catalog.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	private function request_catalog_async( $fields, $force_generate ) {
+		try {
+			$plugin = \Automattic\WooCommerce\ProductFeedForOpenAI\Core\Plugin::get_instance();
+			if ( ! $plugin ) {
+				return new WP_Error(
+					'async_generation_unavailable',
+					__( 'Product Feed plugin instance unavailable.', 'woocommerce' ),
+					array( 'status' => 500 )
+				);
+			}
+
+			$generator = $plugin->get( \Automattic\WooCommerce\ProductFeedForOpenAI\Integrations\POSCatalog\AsyncGenerator::class );
+			if ( ! $generator ) {
```

Comment:
It'd be good to confirm for the Product Feed plugin devs (which I see tagged here), but I don't think this could happen. I believe either the container returns the instance or an exception is thrown.